lib/rucio/core/replica.py

```diff
@@ -799,6 +808,48 @@ def _list_replicas_for_files(file_clause, state_clause, files, rse_clause, updat
             {'scope': scope, 'name': name} in files and files.remove({'scope': scope, 'name': name})
 
 
+def get_vp_endpoint():
+    vp_endpoint = config_get('virtual_placement', 'vp_endpoint', default='')
+    print('vp endpoint: ', vp_endpoint)
+    return vp_endpoint
+
+
+def get_multi_cache_prefix(cache_site, filename, logger=logging.log):
+
+    vp_endpoint = get_vp_endpoint()
+    if not vp_endpoint:
+        return ''
+
+    # print('Looking up prefix for cache:', cache_site, 'and filename:', filename)
+    x_caches = REGION.get('CacheSites')
+    if x_caches is NO_VALUE:
+        logger(logging.DEBUG, 'reloading xcache sites and ranges.')
+        try:
+            response = requests.get('{}/serverRanges'.format(vp_endpoint), verify=False)
+            if response.status_code == 200:
+                x_caches = response.json()
+                logger(logging.DEBUG, x_caches)
+                REGION.set('CacheSites', x_caches)
+            else:
+                return ''
+        except requests.exceptions.RequestException as re:
+            REGION.set('CacheSites', {'could not reload': ''})
+            logger(logging.ERROR, 'In get_multi_cache_prefix, could not access {}. Error:{}'.format(vp_endpoint, re))
+            return ''
+
+    if cache_site not in x_caches:
+        logger(logging.DEBUG, 'cache site: ' + cache_site + ' not active.')
```

Comment:
is this actually a case which is useful when debugging a problem or does this happen often?

Reply:
I would probably remove this. In theory it should not happen too often, but when it happens it is really quite spammy.

Reply:
it could happen that a client was pointing to a cache that had only one cache node and that one went dead. Or have it wrongly configured. In that case it would be spammy. So I removed the printout.